crates/store/re_redap_client/src/grpc.rs

```diff
@@ -324,15 +325,19 @@ pub async fn stream_blueprint_and_segment_from_server(
             store_version: None,
         };
 
-        stream_segment_from_server(
+        if stream_segment_from_server(
             &mut client,
             blueprint_store_info,
             &tx,
             blueprint_dataset,
             blueprint_segment,
             re_uri::Fragment::default(),
         )
-        .await?;
+        .await?
+        .is_break()
+        {
+            return Ok(());
```

Comment:
Small correctness fix: if the receiver is dropped, we would previously continue trying to load the recording etc. Now we stop earlier.